src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceManifest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.CharBuffer;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+import org.opensearch.SpecialPermission;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.DeprecationHandler;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+
+/**
+ * Ip2Geo datasource manifest file object
+ *
+ * Manifest file is stored in an external endpoint. OpenSearch read the file and store values it in this object.
+ */
+@Setter
+@Getter
+@AllArgsConstructor
+public class DatasourceManifest {
+    private static final ParseField URL_FIELD = new ParseField("url");
+    private static final ParseField DB_NAME_FIELD = new ParseField("db_name");
+    private static final ParseField MD5_HASH_FIELD = new ParseField("md5_hash");
+    private static final ParseField VALID_FOR_IN_DAYS_FIELD = new ParseField("valid_for_in_days");
```

Comment:
this field name 'valid for in days ' is not clear to me, can you give me some more context?